### ISSUE TITLE
Combobox and Dropdown hover foreground text colour doens't have sufficient contrast in teamsHighContrast

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#28322]
Combobox:
![image](https://github.com/microsoft/fluentui/assets/20744592/29863f0b-ef06-43be-8b87-faa6633fcef4)

Dropdown:
![image](https://github.com/microsoft/fluentui/assets/20744592/946decc6-714c-4e79-bd6e-d6a7ab28a85e)

The contrast of the foreground text to the hovered background is insufficient, and should behave similarly to the colour pair used in Menu

![image](https://github.com/microsoft/fluentui/assets/20744592/62370eaf-02ae-4bca-a5cc-a217b1f84a5f)

**Comment by dmytrokirpa** [comment]
The issue mentioned in the PR fixed in some other PR for both Combobox and Dropdown components

https://github.com/user-attachments/assets/3a902b7b-cc14-41e2-b63d-b91d3bd571d2
